### PR TITLE
initial teams support

### PR DESCRIPTION
**Pull request by bdistin**
### Description of the PR
This preps the Klasa API for #632 . This finalizes the structure for teams, but without the API endpoint, it won't automatically be filled with the team owners until that is implemented.

This breaks usage of Client#owner and the ownerID client option.

### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)

-

### Semver Classification

- [ ] This PR only includes documentation or non-code changes.
- [ ] This PR fixes a bug and does not change the (intended) framework interface.
- [ ] This PR adds methods or properties to the framework interface.
- [x] This PR removes or renames methods or properties in the framework interface.

### CODE REVIEW

File: src/lib/Client.js [hunk 1]
```diff
@@ -340,13 +340,18 @@ class KlasaClient extends Discord.Client {
...
-	 * @type {?KlasaUser}
+	 * The owners for this bot
+	 * @since 0.5.0
+	 * @type {Set<KlasaUser>}
 	 * @readonly
 	 */
-	get owner() {
-		return this.users.get(this.options.ownerID) || null;
+	get owners() {
+		const owners = new Discord.Collection();
```
Comment (kyranet) [inline src/lib/Client.js]:
```suggestion
		const owners = new Set();
```

File: src/lib/Client.js [hunk 2]
```diff
@@ -340,13 +340,18 @@ class KlasaClient extends Discord.Client {
...
+		for (const owner of this.client.options.owners) {
```
Comment (tech6hutch) [inline src/lib/Client.js]: ```this.options.owners```

File: src/lib/Client.js [hunk 3]
```diff
@@ -340,13 +340,18 @@ class KlasaClient extends Discord.Client {
...
 	/**
-	 * The owner for this bot
-	 * @since 0.1.1
-	 * @type {?KlasaUser}
+	 * The owners for this bot
+	 * @since 0.5.0
+	 * @type {Set<KlasaUser>}
 	 * @readonly
 	 */
-	get owner() {
```
Comment (tech6hutch) [inline src/lib/Client.js]: If you wanted to keep some backwards compatibility, you could do
```js
get owner() {
    return this.options.owners.length === 1 ? this.users.get(this.options.owners.values().next().value) : null;
}
```

Comment (kyranet) [reply src/lib/Client.js]: No, that'd make many things harder, overloaded outputs are a pain to maintain.

File: src/lib/Client.js [hunk 4]
```diff
@@ -340,13 +340,18 @@ class KlasaClient extends Discord.Client {
...
-		return this.users.get(this.options.ownerID) || null;
+	get owners() {
+		const owners = new Set();
+		for (const owner of this.options.owners) {
+			const user = this.users.get(owner);
+			if (user) owners.add(user);
```
Comment (vladfrangu) [inline src/lib/Client.js]: Can't we use partial users here?

Comment (bdistin) [reply src/lib/Client.js]: No, that can be a team id currently.

**Pull request merged**